lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -952,15 +952,16 @@ findFieldWithName(const std::vector<swift::reflection::FieldInfo> &fields,
   return {SwiftLanguageRuntime::eFound, child_indexes.size()};
 }
 
-std::optional<std::string> SwiftLanguageRuntimeImpl::GetEnumCaseName(
+llvm::Expected<std::string> SwiftLanguageRuntimeImpl::GetEnumCaseName(
     CompilerType type, const DataExtractor &data, ExecutionContext *exe_ctx) {
   using namespace swift::reflection;
   using namespace swift::remote;
   auto *ti = GetSwiftRuntimeTypeInfo(type, exe_ctx->GetFramePtr());
   if (!ti)
-    return {};
+    return llvm::createStringError("could not get runtime type info for " +
+                                   type.GetMangledTypeName().GetStringRef());
   if (ti->getKind() != TypeInfoKind::Enum)
-    return {};
+    return "";
```

Comment:
This should probably be an error too no?

Reply:
Surprisingly, perhaps not: There is a weird compiler optimization happening with single-case payload carrying enums (rdar://139521326).